### PR TITLE
Adding host tests

### PR DESCRIPTION
**Pull request by Lydias303**
added a before filter so only a host can see their dashboard. added tests with it.

### CODE REVIEW

File: spec/feature/authenticated_host_spec.rb [hunk 1]
```diff
@@ -54,4 +54,41 @@
...
+    visit root_path
+    click_link_or_button("Dashboard")
+    expect(page).to have_content("Edit Your Listings")
+    expect(page).to have_content("Manage Reservations")
+    expect(page).to have_content("Reservation Requests")
+  end
+
+  it "can view their listings as links to the edit page" do
+    valid_user = create(:user, role: 1)
+    listing = create(:listing)
```
Comment (bryceholcomb) [inline spec/feature/authenticated_host_spec.rb]: Is this factory girl?

File: spec/feature/authenticated_user_spec.rb [hunk 2]
```diff
@@ -257,6 +257,15 @@
     visit new_admin_path
     expect(page).to have_content("Page Not Found")
   end
+
+  it "cannot view a hosts dashboard" do
```
Comment (bryceholcomb) [inline spec/feature/authenticated_user_spec.rb]: Very nice, glad to see some authorization being worked in

File: spec/feature/authenticated_host_spec.rb (hunk 1 above)
Comment (Lydias303) [inline spec/feature/authenticated_host_spec.rb]: yeah I just added role to factory girl and then changed it to 1 in the test to make it a host. I was having some issues when creating a user and a host as it was saying i had repeated email and display name so I had to change those too. If I did it wrong Im more than happy to change it.

File: spec/feature/authenticated_user_spec.rb (hunk 2 above)
Comment (Lydias303) [inline spec/feature/authenticated_user_spec.rb]: :sparkles:

**Comment by cluhring** [comment]
what does your jQuery do?

**Comment by cluhring** [comment]
I, like Bryce also dig the authorizations.  Nice tests :)

File: spec/feature/authenticated_host_spec.rb (hunk 1 above)
Comment (bryceholcomb) [inline spec/feature/authenticated_host_spec.rb]: No ive just never used it, it looks very handy

**Pull request merged**